src/cosmos/cosmosTypes.ts

```diff
@@ -164,3 +174,51 @@ export const asThornodeNetwork = asObject({
   // total_reserve: asString, // '8596638591565880',
   // vaults_migrating: false
 })
+
+/**
+ * WalletConnect payloads
+ */
+export const asCosmosWcGetAccountsRpcPayload = asObject({
+  method: asValue('cosmos_getAccounts'),
+  params: asObject({})
+})
+const asSignDirect = asValue('cosmos_signDirect')
+export const asCosmosWcSignDirectRpcPayload = asObject({
+  method: asSignDirect,
+  params: asObject({
+    signerAddress: asString,
+    signDoc: asObject({
+      chainId: asString,
+      accountNumber: asString,
+      authInfoBytes: asString,
+      bodyBytes: asString
```

Comment:
Could this be `asEither(asBase16, asBase64)`? Then you don't need `safeParse`. Similarly for the other places.